packages/upload-client/src/index.js

```diff
@@ -122,58 +124,68 @@ async function uploadBlockStream(
-            piece = Link.create(raw.code, multihashDigest)
-            const content = Link.create(raw.code, multihash)
+      /** @type {TransformStream<import('./types.js').IndexedCARFile, import('./types.js').CARMetadata>} */
+      (
+        new TransformStream({
+          async transform(car, controller) {
+            const bytes = new Uint8Array(await car.arrayBuffer())
+            // Invoke blob/add and write bytes to write target
+            const multihash = await Blob.add(conf, bytes, options)
+            // Should this be raw instead?
+            const cid = Link.create(carCodec.code, multihash)
+            let piece
+            if (pieceHasher) {
+              const multihashDigest = await pieceHasher.digest(bytes)
+              /** @type {import('@web3-storage/capabilities/types').PieceLink} */
+              piece = Link.create(raw.code, multihashDigest)
+              const content = Link.create(raw.code, multihash)
 
-            // Invoke filecoin/offer for data
-            const result = await Storefront.filecoinOffer(
-              {
-                issuer: conf.issuer,
-                audience: conf.audience,
-                // Resource of invocation is the issuer did for being self issued
-                with: conf.issuer.did(),
-                proofs: conf.proofs,
-              },
-              content,
-              piece,
-              options
-            )
-
-            if (result.out.error) {
-              throw new Error(
-                'failed to offer piece for aggregation into filecoin deal',
-                { cause: result.out.error }
+              // Invoke filecoin/offer for data
+              const result = await Storefront.filecoinOffer(
+                {
+                  issuer: conf.issuer,
+                  audience: conf.audience,
+                  // Resource of invocation is the issuer did for being self issued
+                  with: conf.issuer.did(),
+                  proofs: conf.proofs,
+                },
+                content,
+                piece,
+                options
               )
+
+              if (result.out.error) {
+                throw new Error(
+                  'failed to offer piece for aggregation into filecoin deal',
+                  { cause: result.out.error }
+                )
+              }
             }
-          }
-          const { version, roots, size } = car
-          controller.enqueue({ version, roots, size, cid, piece })
-        },
-      })
+            const { version, roots, size, slices } = car
+            controller.enqueue({ version, roots, size, cid, piece, slices })
+          },
+        })
+      )
     )
     .pipeTo(
       new WritableStream({
         write(meta) {
           root = root || meta.roots[0]
           shards.push(meta.cid)
+
+          // add the CAR shard itself to the slices
+          meta.slices.set(meta.cid.multihash, [0, meta.size])
+          shardIndexes.push(meta.slices)
```

Comment:
Check with [`ShardedDAGIndex`](https://www.google.com/url?q=https://protocol.zoom.us/j/98311855602?pwd%3DamtYZlludmFuVHkvc3hvNFhOMGIwZz09&sa=D&source=calendar&ust=1716168135804736&usg=AOvVaw02Bj84-RrJKZS70kw7i3pX), but I thought this was supposed to be the first slice.

Reply:
They are sorted when encoded anyway so it doesn't matter where you put it?